src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/VsSolutionUpdatesProjectSnapshotChangeTrigger.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.LanguageServices.Razor.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.ProjectSystem
+{
+    [Export(typeof(ProjectSnapshotChangeTrigger))]
+    internal class VsSolutionUpdatesProjectSnapshotChangeTrigger : ProjectSnapshotChangeTrigger
+    {
+        private readonly IServiceProvider _services;
+        private readonly TextBufferProjectService _projectService;
+
+        [ImportingConstructor]
+        public VsSolutionUpdatesProjectSnapshotChangeTrigger(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider services,
+            TextBufferProjectService projectService)
+        {
+            _services = services;
+            _projectService = projectService;
+        }
+
+        public override void Initialize(ProjectSnapshotManagerBase projectManager)
+        {
+            var eventSink = new SolutionUpdateEventSink(projectManager, _projectService);
+
+            // Attach the event sink to solution update events.
```

Comment:
Do we have to worry about unattaching?

Reply:
Don't think so. We only even attach it once. Same as Workspace_Changed events.